src/WebJobs.Script.Scaling/ScaleManager.cs

```diff
@@ -155,18 +156,19 @@ private void OnUpdateWorkerStatus(object state)
         /// </summary>
         protected virtual async Task PingWorker(string activityId, IWorkerInfo worker)
         {
-            var pingResult = true;
-            if (_pingWorkerUtc < DateTime.UtcNow)
+            // if ping was unsuccessful, keep pinging.  this is to address
+            // the issue where site continue to run on an unassigned worker.
+            if (!_pingResult || _pingWorkerUtc < DateTime.UtcNow)
```

Comment:
So even though this PingWorker method takes a worker info, the assumption is that it is only used to ping the worker corresponding to this VM? Otherwise caching the ping results like you're doing wouldn't be correct, right? I.e. your method can ping ANY worker, but you're caching results for the last one you pinged.